app/models/paper.rb

```diff
@@ -43,4 +45,8 @@ def valid_methodologies
       'case report'
     ]
   end
+
+  def find_publication
+    self.publication = Publication.where(name: self.publication.name).first_or_initialize
```

Comment:
What happens if `self.publication` is `nil` and you call `self.publication.name`?